File: examples/Comopt_Kraus_singleparameter.py
import numpy as np
from quanestimation import *
from qutip import *

# Kraus operators for the amplitude damping channel
# initial state
rho0 = 0.5 * np.array([[1.0, 1.0], [1.0, 1.0]])

gamma = 0.1
K1 = np.array([[1.0, 0.0], [0.0, np.sqrt(1 - gamma)]])
K2 = np.array([[0.0, np.sqrt(gamma)], [0.0, 0.0]])
K = [K1, K2]

dK1 = np.array([[1.0, 0.0], [0.0, -0.5 / np.sqrt(1 - gamma)]])
dK2 = np.array([[0.0, 0.5 / np.sqrt(gamma)], [0.0, 0.0]])
dK = [[dK1], [dK2]]

# comprehensive optimization algorithm: PSO
PSO_paras = {
    "particle_num": 10,
    "psi0": [],
    "ctrl0": [],
    "measurement0": [],
    "max_episode": [100, 10],
    "c0": 1.0,
    "c1": 2.0,
    "c2": 2.0,
    "seed": 1234,
}
com = ComprehensiveOpt(savefile=False, method="PSO", **PSO_paras)
com.kraus(K, dK)
com.SM()

# comprehensive optimization algorithm: DE
DE_paras = {
    "popsize": 10,
    "psi0": [],
    "ctrl0": [],
    "measurement0": [],
    "max_episode": 100,
    "c": 1.0,
    "cr": 0.5,
    "seed": 1234,
}
com = ComprehensiveOpt(savefile=False, method="DE", **DE_paras)
com.kraus(K, dK)
com.SM()

File: quanestimation/examples/Julia examples/Comopt_Kraus_single_parameter.jl
using Random
using StableRNGs
include("../src/QuanEstimation.jl")

# Kraus operators for the amplitude damping channel
γ = 0.1
K1 = [1. 0; 0 sqrt(1-γ)]
K2 = [0. sqrt(γ); 0 0]
K = [K1, K2]

dK1 = [1. 0; 0 -sqrt(1-γ)/2]
dK2 = [0. sqrt(γ); 0 0]
dK = [[dK1], [dK2]]

opt = QuanEstimation.SMopt()
dynamics = QuanEstimation.Kraus(opt, K, dK)

# set objective
obj = QuanEstimation.CFIM_Obj()

# control algorithm: PSO
alg = QuanEstimation.PSO(max_episode=[100, 10], p_num=10, c0=1.0, c1=2.0, c2=2.0, rng= MersenneTwister(1234))
QuanEstimation.run(opt, alg, obj, dynamics;savefile=false)

# control algorithm: DE
alg = QuanEstimation.DE(max_episode=100, p_num=10, c=1.0, cr=0.5, rng=MersenneTwister(1234))
QuanEstimation.run(opt, alg, obj, dynamics;savefile=false)
